newrelic/hooks/logger_structlog.py

```diff
@@ -50,16 +47,28 @@ def wrap__process_event(wrapped, instance, args, kwargs):
         settings = global_settings()
 
     # Return early if application logging not enabled
-    if settings and settings.application_logging and settings.application_logging.enabled:
-        if settings.application_logging.local_decorating and settings.application_logging.local_decorating.enabled:
-            event = add_nr_linking_metadata(event)
+    if settings and settings.application_logging.enabled:
+        if isinstance(event, (str, bytes, bytearray)):
+            message = original_message = event
+            event_attrs = {}
+        elif isinstance(event, dict):
+            message = original_message = event.get("event", "")
+            event_attrs = {k: v for k, v in event.items() if k != "event"}
+        else:
+            # Unclear how to proceed, ignore log. Avoid logging an error message or we may incur an infinite loop.
+            return event
+
+        if settings.application_logging.local_decorating.enabled:
+            message = add_nr_linking_metadata(message)
+            if isinstance(event, (str, bytes, bytearray)):
+                event = message
+            elif isinstance(event, dict) and "event" in event:
+                # TODO CHECK ON THIS
```

Comment:
Was this TODO checked?

Reply:
Nope, that definitely needs checked

Reply:
I checked what this was about, it should be being tested.